carbon-design-system/carbon
# feat(react): upgrade Downshift dependency to v5

We were on v1 Downshift and there are quite very meaningful accessibility fixes/improvements made. This PR upgrades downshift and updates tests/code for any breaking changes. 

### included changes
- Downshift updated from v1 to v5
- combobox/tests updated
- dropdown/tests updated
- filterable multiselect/tests updated
- multiselect/tests updated
- test helpers updated with new downshift props

### to test
Open Multiselect, Dropdown, and Combobox and makes sure all functionality is working as expected 🏄 

_@aledavila @abbeyhrt and anyone on core working on our listy-dropdowny components. We should do a quick check on any bugs in our backlog and see if we don't get free fixes via this update 👍_
